### DISCUSSION TITLE
Obstacle Course mod

### DISCUSSION THREAD

**Discussion by Axeanz** [ppy/osu#13073] [Ideas]
A mod that places a solid block at a random location on the screen, forcing the player to move around it. I honestly think it's a pretty good idea, but probably too difficult to add. You never know if you don't give it a try, so here is the idea.

- If you're playing a map that's very long, there is almost no place where the obstacle doesn't block a note directly. Possible solution is to make the block move somewhere else between breaks, or the obstacle very slowly moves through the entire map. Higher OD or AR - Faster moving obstacle; bigger circles, smaller obstacle. 
- If you're playing a very easy map(below 4*) and short map, it won't make it that much more challenging. Possible solution is to set per star-difficulty, a number of obstacles. If the difficulty is between 0-2, there are 4 obstacles, between 2-3.5 there are 3 obstacles, between 3.5-4.5 there are 2 obstacles, 4.5+ there is 1 obstacle.

- The obstacle can have 2 modes, one which will count as a miss when hit; one where it's just a solid object that stops the cursor from going through.
- The obstacle should probably be good to see, so maybe it can be outlined with a brighter colour. If the player is playing with HD, the colour can be less bright or can be blinking.
- If the player is playing with FL, the block should still be visible, or the cursor should have a seperate larger FL-range for the obstacles.
- If the player is playing with traceable, the obstacle doesn't have a brighter outline.
- Wiggle can do two things with the obstacle: Make it move faster; or make it move less direct.
- Autoplay and Autopilot can probably not be set to play like a cursor-dance-bot, but I haven't thought of any other solutions yet.

**Comment by MightyHelper** [comment]
**Difficulty calculation ideas**:
- Make the obstacle's path based off of a random hash of the map, (so that it's the same always for the same map)
- If a jump is interrupted by the obstacle : Calculate difficulty as if extra hit circle at avg time of the 2 and slightly to de side?

**Other suggestions**:
- I would expect as Cs grows, so does the object, meaning, smaller hit circles cause a bigger object.
- I like the idea of it slowly shifting to avoid being on top of a hit circle, or cut a slider.
- Not sure what would happen in the case of a graphics tablet? Maybe something like ( if the obstacle is a square )
  cursorpos // the position of the cursor on the screen
  mousepos // the position that the mouse * should * be in, provided by os.
  if the cursorpos is in the obstacle, create a line that it can't go through until it goes back and out of this "lock", meanwhile you can see the mousepos as a fragment of the cursor skin?

**Other notes**:
When I read the name of the suggestion, I imagined a **new type of map**, like a normal osu! beatmap, but there is like a black mask(probably over 50% of the screen)that for as long as the cursor is over this mask the players hp will drop faster than usual, and might cause the player to fail.
